# Its Own Slope

So we're looking for a function whose value is equal to the size of its slope. One such trivial function would be

f(x) = 0

But are there any more interesting ones

In [9]:
import math

## So what would it look like?

Start out detective work by cutting put a few triangles...


So there are plenty of functions that have a value that match their slope. In fact there are infinite ones.

Add diagram

So what else do we know

If there is a function f(x) that yields its own slope then the function Af(x) would also yield its own slope. Setting A to zero gives the trivial case that is zero everywhere. So let's say f(x) is the function that f(0) = 1

$$f(0) = 1$$

This is the taylor expansion we saw from Wigglometer:

$$f(x) = f(0) + f'(0).x/1! + f''(0).x^2/2! + f'''(0).x^3/3!$$

Given that the slope is the same value as the function if we were to plot the slope then the slope of that slope would also be the same as the function; and the slope of the slope of the slope. That is to say:

$$f(0) \equiv f'(0) \equiv f''(0) \equiv f'''(0) \equiv 1$$

Hence f(x) would be:

$$f(x) = 1 + x/1! + x^2/2! + x^3/3!$$

So f(1) would be:

$$f(1) = 1 + 1/1! + 1/2! + 1/3!$$

As we saw before in the Tug of War these factorials get big very quickly so if this is a finite number we may expect it to appear after not many iterations.

Let's look:


In [10]:
def self_slope_at_one_generator():
    factorial = 0
    summation = 0
    while True:
        summation += 1 / math.factorial(factorial)
        yield summation
        factorial += 1  

In [11]:
gen = self_slope_at_one_generator()
for x in range(20):
    print('Up to {:2} factorial: {}'.format(x, next(gen)))

Up to  0 factorial: 1.0
Up to  1 factorial: 2.0
Up to  2 factorial: 2.5
Up to  3 factorial: 2.6666666666666665
Up to  4 factorial: 2.708333333333333
Up to  5 factorial: 2.7166666666666663
Up to  6 factorial: 2.7180555555555554
Up to  7 factorial: 2.7182539682539684
Up to  8 factorial: 2.71827876984127
Up to  9 factorial: 2.7182815255731922
Up to 10 factorial: 2.7182818011463845
Up to 11 factorial: 2.718281826198493
Up to 12 factorial: 2.7182818282861687
Up to 13 factorial: 2.7182818284467594
Up to 14 factorial: 2.71828182845823
Up to 15 factorial: 2.718281828458995
Up to 16 factorial: 2.718281828459043
Up to 17 factorial: 2.7182818284590455
Up to 18 factorial: 2.7182818284590455
Up to 19 factorial: 2.7182818284590455


In [12]:
math.e

2.718281828459045

So what do we know about the function so far? We know...

1. It exists (we can draw it).

2. We know the value at f(0)
$$f(0) = 1$$

3. As it goes to the right of the origin it goes up very quickly with small increments in x.

4. To the left of the origin it goes down very slowly approaching zero but never quite reaching it.

5. We know the value of f(1)$$f(1) \approx 2.71828$$

6. We can express it as a polynomial:$$f(x) = 1 + x/1! + x^2/2! + x^3/3!$$

7. I don't quite remember how we can infer this; but:
$$f(x + y) \equiv f(x)f(y)$$



Compare the above with the exponentials we saw in Tug of War.

1. The shape looks similar to the one we would expect.
2. All exponentials are one as x = 0
3. They go up very quickly to the right
4. And slowly approach zero to the left
5. They have the additive property
$$10^(2+5) = 10^2.10^5$$
6. If it goes through x^1 it is e so if it is an exponential function it would be
$$f(x) = e^x$$

## Proof

Let's start with our definition of the slope from ...

$$slope of e^x = \lim_{dx\to0} e^{x+dx} - e^x / dx$$